src/fields_parse.rs

```diff
@@ -67,15 +67,20 @@ pub fn impl_parse_args_named(
 
 fn create_parser<'a>(
     parser_type: &ParserType,
-    types: impl Iterator<Item = &'a Type>,
+    mut types: impl Iterator<Item = &'a Type>,
     count_args: usize,
 ) -> quote::__private::TokenStream {
     let function_to_parse = match parser_type {
-        ParserType::Default => match count_args {
-            1 => {
-                quote! { (|s: String| Ok((s,))) }
+        ParserType::Default => match types.next() {
```

Comment:
We must check that we have only one argument. In the situation where we get 2 types instead of 1 your code return invalid code, because `types.next()` on `[String, i32]` return `Some(String)`. Please return the check for the `count_args` variable.

Reply:
Oh damn, you're right. Somehow I only thought about having 0 or 1 types, but forgot that a user might pass multiple. Will change tomorrow.